packages/front-end/components/Settings/EditDataSource/ExperimentAssignmentQueries/AddEditExperimentAssignmentQueryModal.tsx

```diff
@@ -124,35 +187,54 @@ export const AddEditExperimentAssignmentQueryModal: FC<EditExperimentAssignmentQ
               required
               {...form.register("userIdType")}
             />
-
+            <StringArrayField
+              label="Dimension Columns"
+              value={userEnteredDimensions}
+              onChange={(dimensions) => {
+                form.setValue("dimensions", dimensions);
+              }}
+            />
             <div className="row">
               <div className="col">
-                <CodeTextArea
+                <Field
                   label="SQL Query"
-                  required
-                  language="sql"
-                  value={userEnteredQuery}
-                  setValue={(sql) => form.setValue("query", sql)}
-                />
-                <div className="form-group">
-                  <label className="mr-2">
-                    Use Name Columns
-                    <Tooltip body="Enable this if you store experiment/variation names as well as ids in your table" />
-                  </label>
-                  <Toggle
-                    id="exposure-query-toggle"
-                    value={userEnteredHasNameCol}
-                    setValue={(hasNameCol) => {
-                      form.setValue("hasNameCol", hasNameCol);
-                    }}
-                  />
-                </div>
-
-                <StringArrayField
-                  label="Dimension Columns"
-                  value={userEnteredDimensions}
-                  onChange={(dimensions) => {
-                    form.setValue("dimensions", dimensions);
+                  render={() => {
+                    return (
+                      <>
+                        <div className="d-flex justify-content-between align-items-center p-1 border rounded">
+                          <button
+                            className="btn btn-sm btn-primary m-1"
+                            onClick={(e) => {
+                              e.preventDefault();
+                              handleTestQuery();
+                            }}
+                          >
+                            <span className="pr-2">
+                              <FaPlay />
+                            </span>
+                            Test Query
+                          </button>
+                          <div className="d-flex m-1">
+                            <label className="mr-2 mb-0">
+                              Use Name Columns
+                            </label>
+                            <input
+                              type="checkbox"
+                              id="exposure-query-toggle"
```

Comment:
You should set the `htmlFor` attribute on the label so that clicking the label toggles the checkbox.  Checkboxes are small and hard to click on their own.